src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +273,475 @@ define(function (require, exports, module) {
+            prefItemType   = _getPrefType(prefItem),
+            hasKeys        = false,
+            tabIndents     = "",
+            numKeys        = 0;
+
+        // Generate the indentLevel string
+        for (iLevel = 0; iLevel < indentLevel; iLevel++) {
+            tabIndents += "\t";
+        }
+
+        // Check if the preference is an object.
+        if (_getPrefType(prefItem) === "object") {
+            prefItemKeys = _getChildPrefs(prefItem);
+            if (Object.keys(prefItemKeys).length > 0) {
+                hasKeys = true;
+            }
+        }
+
+        // There are some properties like "highlightMatches" that
+        // are declared as boolean type but still can take object keys.
+        // The below condition check can take care of cases like this.
+        if (prefItemType !== "object" && hasKeys === false) {
+            return _formatBasicPref(prefItem, prefName, tabIndents);
+        }
+
+        // Indent the beginning of the object.
+        tabIndents += "\t";
+
+        if (prefItemDesc && prefItemDesc.length > 0) {
+            entireText = tabIndents + "// " + prefItemDesc + "\n";
+        }
+
+        entireText += tabIndents + "\"" + prefName + "\": " + "{";
+
+        if (prefItemKeys) {
+            numKeys = Object.keys(prefItemKeys).length;
+        }
+
+        // In case the object array is empty
+        if (numKeys <= 0) {
+            entireText += "}";
+            return entireText;
+        } else {
+            entireText += "\n";
+        }
+
+        // Now iterate through all the keys
+        // and generate nested formatted objects.
+
+        for (property in prefItemKeys) {
+
+            if (prefItemKeys.hasOwnProperty(property)) {
+
+                var pref = prefItemKeys[property];
+
+                if (_isValidPref(pref)) {
+
+                    var formattedText = "";
+
+                    if (_getPrefType(pref) === "object") {
+                        formattedText = _formatPref(property, pref, indentLevel + 1);
+                    } else {
+                        formattedText = _formatBasicPref(pref, property, tabIndents);
+                    }
+
+                    if (formattedText.length > 0) {
+                        entireText += formattedText + ",\n\n";
+                    }
+                }
+            }
+        }
+
+        // Strip ",\n\n" that got added above, for the last property
+        if (entireText.length > 0) {
+            entireText = entireText.slice(0, -3) + "\n" + tabIndents + "}";
+        } else {
+            entireText = "{}";
+        }
+
+        return entireText;
+    }
+
+    function _getDefaultPreferencesString() {
+
+        var property,
+            allPrefs       = PreferencesManager.getAllPreferences(),
+            headerComment  = Strings.DEFAULT_PREFERENCES_JSON_HEADER_COMMENT + "\n\n{\n",
+            entireText     = "";
+
+        for (property in allPrefs) {
```

Comment:
Better use `Object.keys(allPrefs).forEach(function (pref) {})` Here. Sorting the preferences here would also be nice.